### PR TITLE
Pre-Equipped Squad Roles and Fixes

### PR DESCRIPTION
**Pull request by XSlayer300**
## About The Pull Request
As per title, this PR adds pre-equipped squad roles for admin events, along with a portable backpack vendor to select their weapons and kits (barring Squad Leader). I do not have plans for them to become an ERT though, but if this gets enough requests, I'll do it.

This also fixes some bugs relating to PMCs carrying 4 energy bars instead of the usual 2, since my other PR removes the 4 slot limit for helmets. This PR also adds variants for belts, pouches and weapons to be used for the squad roles.

## Why It's Good For The Game
Allows admins to quickly set up admin events relating to marines, without the hassle of us using vendors and equipping the marines individually. This also allows some freedom to choose the kit that the player wants.

## Changelog
:cl:
add: Added Pre-Equipped Squad roles for admins.
admin: Pre-Equipped Squad roles are indicated by "(Equipped)" after their name.
add: Added new variants for belts, pouches and weapons for use of Pre-Equipped Squad roles.
fix: Squashed a few unseen bugs of code.
/:cl:

### CODE REVIEW

File: code/game/objects/items/portable_vendor.dm [hunk 1]
```diff
@@ -211,3 +211,75 @@
 							list("De Void of Soul", 51, /obj/item/clothing/under/liaison_suit/galaxy_blue, "white", "The latest in ultrafashion. for those with a cool demeanor."),
 							list("Pulsar Gonne", 51, /obj/item/clothing/under/liaison_suit/galaxy_red, "white", "The latest in ultrafashion. for those with a fiery temper.")
 							)
+
+// A marine version of the portable vendor, points are not regenerated.
+/obj/item/portable_vendor/marine
```
Comment (spookydonut) [inline code/game/objects/items/portable_vendor.dm]: i'm not a fan of how this has been done.
having a physical item for this feels pretty wrong gameplay wise, especially given it doesnt stop you keeping it as a pocket resupply, doesnt self delete when empty and isnt locked to the marine it was given to.

A verb might be better imo.

Comment (XSlayer300) [reply code/game/objects/items/portable_vendor.dm]: I can do the latter two though. Though I don't really understand the first one, like... you can simply keep this item to quickly resupply yourself? It literally has one point of usage and points cannot be regenerated.

`/obj/item/portable_vendor/marine/process()
	points = min(max_points, points+0)`

Comment (spookydonut) [reply code/game/objects/items/portable_vendor.dm]: You can scavenge a gun somewhere and keep this in your pocket for a free resupply

File: code/modules/projectiles/guns/shotguns.dm [hunk 2]
```diff
@@ -761,3 +761,11 @@ can cause issues with ammo types getting mixed up during the burst.
...
 	recoil_unwielded = 4
 	pump_delay = 14
+
+//buckshot variants
+/obj/item/weapon/gun/shotgun/pump/t35/pointman
+	current_mag = /obj/item/ammo_magazine/internal/shotgun/pump/buckshot
+
+/obj/item/weapon/gun/shotgun/pump/t35/nonstandard
+	current_mag = /obj/item/ammo_magazine/internal/shotgun/pump/buckshot
+	starting_attachment_types = list(/obj/item/attachable/stock/t35stock, /obj/item/attachable/angledgrip, /obj/item/attachable/magnetic_harness)
```
Comment (spookydonut) [inline code/modules/projectiles/guns/shotguns.dm]:
```suggestion
	starting_attachment_types = list(/obj/item/attachable/stock/t35stock, /obj/item/attachable/angledgrip, /obj/item/attachable/magnetic_harness)

```

**Comment by XSlayer300** [comment]
Other than fixing minor details or adding some spices to some features, I feel like I have lost the urge to continue the project in general. I hope someone can pick this up and continue with it.

File: code/game/objects/items/portable_vendor.dm [hunk 3]
```diff
@@ -211,3 +211,75 @@
...
+							list("Sniper (M42A Scoped Rifle, 88 Mod 4 Pistol)", 1, /obj/item/storage/box/squadmarine/sniper, "white", "A kit specializing in long-ranged dispatch."),
+							list("Scout (M4RA Battle Rifle, 88 Mod 4 Pistol)", 1, /obj/item/storage/box/squadmarine/scout, "white", "A kit specializing in area scouting."),
+							list("Tracker (MBX900 Lever-Action Shotgun, M1911 Pistol)", 1, /obj/item/storage/box/squadmarine/tracker, "white", "A kit specializing in chasing enemy targets."),
+							list("Pyrotechician (M240T Incinerator, T-19 SMG)", 1, /obj/item/storage/box/squadmarine/pyro, "white", "A kit specializing in firey area denial."),
+							list("Heavy Grenadier (M92 Grenade Launcher)", 1, /obj/item/storage/box/squadmarine/heavy_grenadier, "white", "A kit specializing in explosive area denial."),
+							list("Heavy Minigunner (MIC-A7 Minigun, M4A3 Pistol)", 1, /obj/item/storage/box/squadmarine/heavy_gunner, "white", "A kit specializing in direct assault and tanking.")
+							)
+
+/obj/item/portable_vendor/marine/process()
+	points = min(max_points, points+0)
```
Comment (psykzz) [inline code/game/objects/items/portable_vendor.dm:285]:
```suggestion
							)

/obj/item/portable_vendor/marine/process()
	STOP_PROCESSING(SSobj, src)
	return FALSE

/obj/item/portable_vendor/marine/do_vend()
	. = ..()
	STOP_PROCESSING(SSobj, src)
	qdel(src)
```

File: code/datums/jobs/job/marines.dm [hunk 4]
```diff
@@ -88,8 +134,7 @@ Your squaddies will look to you when it comes to construction in the field of ba
 	job_flags = JOB_FLAG_LATEJOINABLE|JOB_FLAG_ROUNDSTARTJOINABLE|JOB_FLAG_ALLOWS_PREFS_GEAR|JOB_FLAG_PROVIDES_BANK_ACCOUNT|JOB_FLAG_ADDTOMANIFEST|JOB_FLAG_PROVIDES_SQUAD_HUD
 	jobworth = list(/datum/job/xenomorph = LARVA_POINTS_REGULAR, /datum/job/terragov/squad/specialist = SPEC_POINTS_MEDIUM, /datum/job/terragov/squad/smartgunner = SMARTIE_POINTS_MEDIUM)
 
-
-/datum/job/terragov/squad/corpsman/radio_help_message(mob/M)
+/datum/job/marine/corpsman/radio_help_message(mob/M)
```
Comment (psykzz) [inline code/datums/jobs/job/marines.dm:137]: Can you double check these type paths?

File: code/datums/jobs/job/marines.dm [hunk 5]
```diff
@@ -73,6 +94,31 @@ Your squaddies will look to you when it comes to construction in the field of ba
...
 
+/datum/outfit/job/marine/engineer/equipped
+	name = "Squad Engineer (Equipped)"
+
+	belt = /obj/item/storage/belt/utility/full
+	ears = /obj/item/radio/headset/mainship/marine
+	w_uniform = /obj/item/clothing/under/marine/engineer
+	wear_suit = /obj/item/clothing/suit/storage/marine
+	shoes = /obj/item/clothing/shoes/marine
+	gloves =/obj/item/clothing/gloves/marine
```
Comment (Mark9013100) [inline code/datums/jobs/job/marines.dm:105]: recommend you replace the gloves line with this: gloves = /obj/item/clothing/gloves/marine/insulated

**Comment by Timmeye** [comment]
**s p e e d  m e r g e**

**Comment by psykzz** [comment]
This still has outstanding changes that need addressing

File: code/datums/jobs/job/marines.dm (hunk 4 above)
Comment (XSlayer300) [reply code/datums/jobs/job/marines.dm]: Checked it now, I suppose.

**Comment by XSlayer300** [comment]
Waiting on review approval.

**Comment by terranaut1** [comment]
does not compile

**Comment by XSlayer300** [comment]
The UI for the backpack only appears when you are at the RIGHT JOB CLASS (As in, you cannot use the medic portable vendor when you are a squad engineer and vice versa. You can only use medic portable vendors when you are a medic ONLY).

**Comment by psykzz** [comment]
I spawned in as a human, used the job panel to equip and set my role.
I did squad specialist at first and noticed my mistake before trying again as corpsman, both failed, with no message.

That said would be able to add an error message to the bag incase one does go to the wrong person?

**Comment by XSlayer300** [comment]
It's kind of dirty, but I got the job done.

**Comment by psykzz** [comment]
Seems to be working a lot better, though i have some final comments.

If you could add some text about vending a box, at first i didn't know what to expect when i clicked the button until a box spawned a few seconds later (because vending is slow)

Smartgunner doesn't spawn with any ammo, and doens't need the M56 powerpack anymore.
Should add some `/obj/item/ammo_magazine/standard_smartmachinegun` in its place.

I had a cursory look over the rest but it seems fine mostly.

**Pull request merged**